### PR TITLE
Fix tensorboard interval in validation

### PR DESCRIPTION
**Pull request by ShigekiKarita**
After this PR #966 (make tensorboard log every 100 iters), we found that iter 100 is too large to display validation stats. This PR aims to revert the validation TB log interval to 1.

This PR realize this by calling TB logger manually inside the new evaluator (espnet.utils.training.evaluator.BaseEvaluator).

- [x] BaseEvaluator
- [x] use args.report_interval_iters intead of the constant REPORT_INTERVAL. It is consistent to tts_train.py
- [x] add test

### CODE REVIEW

**Comment by ShigekiKarita** [comment]
@kan-bayashi I feel we can merge ASR and TTS CustomEvaluator/Updater. It seems to be same. Do you agree with it?

**Comment by ShigekiKarita** [comment]
Now Tensorboard on AN4 seems to display validation set logs
image

**Comment by kan-bayashi** [comment]
Could you check acc.png is correctly working?

**Comment by kan-bayashi** [comment]
> @kan-bayashi I feel we can merge ASR and TTS CustomEvaluator/Updater. It seems to be same. Do you agree with it?

I will check it lator.

**Comment by ShigekiKarita** [comment]
this is cer.png because acc.png is not generated by mtlalpha=1.0 in the AN4 default
cer

**Comment by ShigekiKarita** [comment]
> I will check it lator.

I think the unification is a necessary step for the input/output data format in the future #1012

**Comment by kan-bayashi** [comment]
> this is cer.png because acc.png is not generated by mtlalpha=1.0 in the AN4 default

Thanks. It looks fine.
Recently, I met strange phenomena in TTS as follows:


スクリーンショット 2019-08-07 午後9 51 16

Evaluation was conducted every 4 epochs, not every epoch.
This might be a bug of TTS.

**Comment by kan-bayashi** [comment]
> I think the unification is a necessary step for the input/output data format in the future #1012

I checked the code and as you mentioned, they are the almost same.
Let us unify them.

**Comment by ShigekiKarita** [comment]
Hmm. I'm running AN4 tts but the all_loss.png seems to be OK. Did you test it LJSpeech?
all_loss

>  Let us unify them.

Yeah. We should split the PR for the unification

**Comment by kan-bayashi** [comment]
>  Did you test it LJSpeech?

I test it on LibriTTS.
The reason may be the use of `ngpu=3` and `accum_grad=8`, but I’m not sure.

File: egs/an4/asr1/local/check_tblog.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env python3
+# coding: utf-8
+
```
Comment (kan-bayashi) [inline egs/an4/asr1/local/check_tblog.py]: Is this file needed?

Comment (ShigekiKarita) [reply egs/an4/asr1/local/check_tblog.py]: no. I made this just for pre-testing.

Comment (ShigekiKarita) [reply egs/an4/asr1/local/check_tblog.py]: no. I made this just for pre-testing.

**Comment by ShigekiKarita** [comment]
Yes. I added the test.

**Comment by sw005320** [comment]
Many thanks!
